build/localnet/cluster-manager/sts_kill.go

```
@@ -0,0 +1,115 @@
+	podList, _ := podClient.List(context.TODO(), metav1.ListOptions{})
+	for i := range podList.Items {
+		pod := podList.Items[i]
+		err := deleteCrashedPods(&pod, stsClient, podClient)
+		if err != nil {
+			logger.Error().Err(err).Msg("error deleting crashed pod on init")
+		}
+	}
+
+	// Set up a watch for new Pods
+	w, err := podClient.Watch(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		logger.Error().Err(err).Msg("error setting up watch for new pods")
+	}
+	for event := range w.ResultChan() {
+		switch event.Type {
+		case watch.Added, watch.Modified:
+			pod, ok := event.Object.(*corev1.Pod)
+			if !ok {
+				continue
+			}
+
+			err := deleteCrashedPods(pod, stsClient, podClient)
+			if err != nil {
+				logger.Error().Err(err).Msg("error deleting crashed pod on watch")
+			}
+		}
+	}
+}
+
+func deleteCrashedPods(pod *corev1.Pod, stsClient appstypedv1.StatefulSetInterface, podClient coretypedv1.PodInterface) error {
+	// If annotation is present, we monitor the Pod
+	if containerToMonitor, ok := pod.Annotations["cluster-manager-delete-on-crash-container"]; ok {
+		for ci := range pod.Spec.Containers {
+			podContainer := pod.Spec.Containers[ci]
+			if podContainer.Name == containerToMonitor {
+				for pi := range pod.Status.ContainerStatuses {
+					containerStatus := pod.Status.ContainerStatuses[pi]
+
+					// Only proceed if container is in some sort of Err status
+					if containerStatus.State.Waiting != nil &&
+						(strings.HasPrefix(containerStatus.State.Waiting.Reason, "Err") ||
+							strings.HasSuffix(containerStatus.State.Waiting.Reason, "BackOff")) {
+						// Get StatefulSet that created the Pod
+						var stsName string
+						for _, ownerRef := range pod.OwnerReferences {
+							if ownerRef.Kind == "StatefulSet" {
+								stsName = ownerRef.Name
+								break
+							}
+						}
+
+						if stsName == "" {
+							return errors.New("no StatefulSet found for this pod")
```

Do we want to return an error or just log it and continue?

---

@Olshansk realistically, this should never happen, so I'd like to see the error if it does.